src/ai/backend/test/testcases/model_service/scale_by_auto_scaling_rule.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+from typing import override
+from uuid import UUID
+
+from ai.backend.client.session import AsyncSession
+from ai.backend.test.contexts.client_session import ClientSessionContext
+from ai.backend.test.contexts.model_service import (
+    AutoScalingRuleContext,
+    CreatedModelServiceEndpointMetaContext,
+    ModelServiceContext,
+)
+from ai.backend.test.templates.model_service.utils import wait_until_all_inference_sessions_ready
+from ai.backend.test.templates.template import TestCode
+from ai.backend.test.templates.vfolder.utils import get_vfolder_id_by_name
+from ai.backend.test.utils.exceptions import DependencyNotSet
+
+_SCALE_TIMEOUT = 30
+
+
+class ScaleByAutoScalingRules(TestCode):
+    @override
+    async def test(self) -> None:
+        endpoint_meta = CreatedModelServiceEndpointMetaContext.current()
+        service_id = endpoint_meta.service_id
+        client_session = ClientSessionContext.current()
+        model_service_dep = ModelServiceContext.current()
+        auto_scaling_rule_dep = AutoScalingRuleContext.current()
+        max_replicas = auto_scaling_rule_dep.max_replicas
+        if max_replicas is None:
+            raise DependencyNotSet("AutoScalingRuleContext.max_replicas must be set")
+        vfolder_id = await get_vfolder_id_by_name(
+            client_session, model_service_dep.model_vfolder_name
+        )
+
+        await self._make_cpu_intensive_task_in_service(
+            client_session,
+            service_id,
+            code="while True: pass",
+        )
+
+        await asyncio.sleep(5)  # Give some time for making cpu-intensive environment
```

Comment:
If possible, please avoid including logic that waits for an arbitrary amount of time.
No matter what arbitrary time is set, it could be too long or too short depending on the situation.
Use polling for now.